server/src/main/java/nl/inl/blacklab/server/requesthandlers/RequestHandler.java

```diff
@@ -797,16 +797,22 @@ public static String getDocumentPid(BlackLabIndex index, int luceneDocId, Docume
         }
     }
 
-    protected void addNumberOfResultsSummaryTotalHits(DataStream ds, ResultsStats hitsStats, ResultsStats docsStats, boolean countFailed, CorpusSize subcorpusSize) {
+    protected void addNumberOfResultsSummaryTotalHits(DataStream ds, ResultsStats hitsStats, ResultsStats docsStats, boolean waitForTotal, boolean countFailed, CorpusSize subcorpusSize) {
         // Information about the number of hits/docs, and whether there were too many to retrieve/count
         // We have a hits object we can query for this information
+
+        long hitsCounted = countFailed ? -1 : (waitForTotal ? hitsStats.countedTotal() : hitsStats.countedSoFar());
```

Comment:
Can  hitStats and docStats be null here?

Reply:
I don't think so, but it can't hurt to be defensive about it. Adding null checks.